src/main/java/com/google/api/gax/grpc/ServiceApiSettings.java

```diff
@@ -238,23 +241,26 @@ public Builder setClientLibHeader(String name, String version) {
     /**
      *  Performs a merge, using only non-null fields
      */
-    @SuppressWarnings("unused")
-    public Builder applyToAllApiMethods(ApiCallSettings.Builder apiCallSettings) throws Exception {
-      // Subclasses need to override
-      throw new Exception("applyToAllApiMethods is not implemented");
-    }
-
-    /**
-     *  Performs a merge, using only non-null fields
-     */
-    @SuppressWarnings("unused")
-    public Builder applyToAllApiMethods(RetrySettings.Builder retrySettings) throws Exception {
-      // Subclasses need to override
-      throw new Exception("applyToAllApiMethods is not implemented");
+    protected Builder applyToAllApiMethods(Iterable<ApiCallSettings.Builder> methodSettingsBuilders,
+        ApiCallSettings.Builder newSettingsBuilder) throws Exception {
+      Set<Status.Code> retryableCodes = newSettingsBuilder.getRetryableCodes();
+      RetrySettings.Builder retrySettingsBuilder = newSettingsBuilder.getRetrySettingsBuilder();
+      for (ApiCallSettings.Builder settingsBuilder : methodSettingsBuilders) {
+        if (retryableCodes != null) {
+          settingsBuilder.setRetryableCodes(retryableCodes);
+        }
+        if (retrySettingsBuilder != null) {
+          settingsBuilder.setRetrySettingsBuilder(retrySettingsBuilder);
```

Comment:
Hmm I had in mind something different. I realized an issue with my suggestion - deep copy is not actually what we want; we want this apply function to actually do a merge, not a deep copy. What that means is that only values which are non-null should be copied; null values in the source builder object should be ignored.